tests/PHPStan/Analyser/nsrt/preg_match_shapes.php

```diff
@@ -518,6 +518,12 @@ function bug11323(string $s): void {
 	if (preg_match('{([^1-4])}', $s, $matches)) {
 		assertType('array{string, non-empty-string}', $matches);
 	}
+	if (preg_match("{([\r\n]+)(\n)([\n])}", $s, $matches)) {
+		assertType('array{string, non-empty-string, non-empty-string, non-empty-string}', $matches);
+	}
+	if (preg_match('/foo(*:first)|bar(*:second)([x])/', $s, $matches)) {
+		assertType("array{0: string, 1?: non-empty-string, MARK?: 'first'|'second'}", $matches);
```

Comment:
should MARK be non-optional since we are sure here about the successfull match?

Reply:
I left it as such because it depends where the mark is https://3v4l.org/vJofm - and it seems to me not really worth the effort to ensure whether it's in a match group that is optional or not, given how niche this feature is.